File: Java/Language/Basics/AbstractClassTest.java
import banking.*;

class AbstractClassTest {

	public static void main(String[] args) {
		Account jack = Banker.openCurrentAccount();
		jack.deposit(10000);
		Account jill = Banker.openSavingsAccount();
		jill.deposit(5000);
		double payment = Double.parseDouble(args[0]);
		try{
			jill.transfer(payment, jack);
			System.out.println("Transfer succeeded.");
		}catch(InsufficientFundsException e){
			System.out.println("Transfer failed!");
		}
		System.out.printf("Jack's Account ID is %d and Balance is %.2f%n", jack.getId(), jack.getBalance());
		System.out.printf("Jill's Account ID is %d and Balance is %.2f%n", jill.getId(), jill.getBalance());
	}
}

File: CSharp/Basics/AbstractClassTest.cs
using Banking;
using System;

static class Program
{
	public static void Main(string[] args)
	{
		Account jack = Banker.OpenCurrentAccount();
		jack.Deposit(15000);
		Account jill = Banker.OpenSavingsAccount();
		jill.Deposit(10000);
		try
		{
			double payment = double.Parse(args[0]);
			jill.Transfer(payment, jack);
			Console.WriteLine("Transfer succeeded.");
		}
		catch(InsufficientFundsException)
		{
			Console.WriteLine("Transfer failed!");
		}
		catch(Exception ex)
		{
			Console.WriteLine("Error: {0)", ex.Message);
		}
		Console.WriteLine("Jack Account ID is {0} and Balance is {1}", jack.Id, jack.Balance);
		Console.WriteLine("Jill Account ID is {0} and Balance is {1}", jill.Id, jill.Balance);
	}
}

